RELEASE_NOTES.md

```diff
@@ -2,15 +2,15 @@
 
 ## Summary
 
-<!-- Here goes a general summary of what this release is about -->
+The `Timer` now can be started with a delay.
 
 ## Upgrading
 
 <!-- Here goes notes on how to upgrade from previous versions, including deprecations and what they should be replaced with -->
 
 ## New Features
 
-<!-- Here goes the main new features and examples or instructions on how to use them -->
+* `Timer()`, `Timer.timeout()`, `Timer.periodic()` and `Timer.reset()` now take an optional `start_delay` option to make the timer start after some delay.
```

Comment:
Should we also briefly mention here the main reason why a user might need to use a delay in Timer?
Also you might want to clarify in the commit message the "why" this feature is needed or convenient to use on top of what it was done.